### ISSUE TITLE
Long wait until UnreachableBrowserException (Firefox)

### ISSUE THREAD

**Issue by awallat** [serenity-bdd/serenity-core#28]
While my tests in Chrome run fine, I have sometimes problems with Firefox. Last time it happened between to test starts (I configured my Maven project with profiles to run only specific tests). After the end of the first test suite, the second started with:
```
org.openqa.selenium.remote.UnreachableBrowserException: Error communicating with the remote browser. It may have died.
Build info: version: '2.44.0', revision: '76d78cf323ce037c5f92db6c1bba601c2ac43ad8', time: '2014-10-23 13:11:40'
System info: host: 'WinGridTest', ip: '192.168.0.250', os.name: 'Windows 8.1', os.arch: 'amd64', os.version: '6.3', java.version: '1.8.0_05'
Driver info: driver.version: RemoteWebDriver
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:593)
        ...
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
```

I'm not sure if this is just Selenium related, but it **took exactly 3h** until the Exception was thrown. I have a timeout configured for 180s (story.timeout.in.secs) ... could this have something to do with it and somehow been transformed to minutes?

**Comment by wakaleo** [comment]
I don't think the timeout configuration is related - that is just passed to
JBehave. It sounds more like a webdriver/firefox issue, especially if it
works in Chrome.

On 20 February 2015 at 21:45, André Wallat <notifications@github.com> wrote:

> While my tests in Chrome run fine, I have sometimes problems with Firefox.
> Last time it happened between to test starts (I configured my Maven project
> with profiles to run only specific tests). After the end of the first test
> suite, the second started with:
>
> org.openqa.selenium.remote.UnreachableBrowserException: Error communicating with the remote browser. It may have died.
> Build info: version: '2.44.0', revision: '76d78cf323ce037c5f92db6c1bba601c2ac43ad8', time: '2014-10-23 13:11:40'
> System info: host: 'WinGridTest', ip: '192.168.0.250', os.name: 'Windows 8.1', os.arch: 'amd64', os.version: '6.3', java.version: '1.8.0_05'
> Driver info: driver.version: RemoteWebDriver
>     at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:593)
>         ...
> Caused by: java.net.SocketTimeoutException: Read timed out
>     at java.net.SocketInputStream.socketRead0(Native Method)
>
> I'm not sure if this is just Selenium related, but it *took exactly 3h*
> until the Exception was thrown. I have a timeout configured for 180s
> (story.timeout.in.secs) ... could this have something to do with it and
> somehow been transformed to minutes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/serenity-bdd/serenity-core/issues/28>.
>



-- 
___________________________________________________
John Smart | Wakaleo Consulting  |  +61 407 247 642
Optimizing your software development process
http://www.wakaleo.com  |  john.smart@wakaleo.com
___________________________________________________

The dates for the 2014 BDD workshops have been scheduled! Check out our
upcoming *BDD/TDD Master classes*
<http://wakaleo.com/training/testing-and-tdd-for-java-developers> and
our *Advanced
BDD Requirements Workshops
<http://wakaleo.com/training/advanced-agile-requirements-workshop>*, coming
soon to *Sydney
<http://wakaleo.com/training/testing-and-tdd-for-java-developers/public-courses>
*
and *Melbourne
<http://wakaleo.com/training/testing-and-tdd-for-java-developers/public-courses>!*
___________________________________________________

**Comment by awallat** [comment]
Is it possible to have a timeout for each test, where the browser will be closed/killed? During my last tests Firefox hang for a couple of hours until I closed the browser manually. Then it continued with the tests.

**Comment by wakaleo** [comment]
Possibly. What testing framework (JUnit, JBehave, Cucumber) are you using? What version of Firefox are you using? It sounds like the browser or webdriver connection hanging, in which case a timeout might not be of much help.

**Comment by wakaleo** [comment]
Sounds like a Firefox Driver issue that was fixed in Selenium 2.45 - could you reopen if you are still having the issue in Serenity 1.0.37?

**Comment by marcosscriven** [comment]
Found this looking for socket hanging issues with selenium/httpclient/firefox webdriver.

What I can tell you is where the @awallat is where the three hours comes from - Selenium sets the socket timeout to that here: https://github.com/SeleniumHQ/selenium/blob/selenium-2.42.2/java/client/src/org/openqa/selenium/remote/internal/HttpClientFactory.java#L119
